webPageFetcher.sh

```diff
@@ -0,0 +1,524 @@
+# Description: Prompts the user to select the     #
+# desired choice from the options.                #
+#                                                 #
+# Arguments: None                                 #
+#                                                 #
+# Returns: None                                   #
+###################################################
+function web_page_chooser()
+{
+  # variable to return the result
+  local user_choice=""
+
+  # prompt user to choose
+  while [[ true ]]; do
+    read -p "$( echo -e "[${LIGHT_BLUE}INFO${NC}] Which Web Page Do You Want"\
+     "To Use [1-$NUMBER_OF_FOLDERS]: ")"
+    echo ""
+
+    # check if given input is integer
+    if [[ "$REPLY" =~ ^[0-9]+$ ]]; then
+      # check if given input is in range
+      if [[ "$REPLY" -ge 1 && "$REPLY" -le $NUMBER_OF_FOLDERS ]]; then
+        # save users choice
+        user_choice="$(( $REPLY-1))"
+
+        # prompt user to confirm choice
+        read -p "$( echo -e "[${LIGHT_BLUE}INFO${NC}] You Have Choosen"\
+        "${FOLDERS[$(( REPLY - 1 ))]}? [Y/n]" )"
+        echo ""
+        if [[ "$REPLY" =~ ^[Yy]$ ]]; then
+        # return choice for next stage
+        return $user_choice
+        fi
+      else
+        echo -e "[${RED}WARNING${NC}] Given Input Is Not In Range!"
+        echo ""
+      fi
+    else
+      echo -e "[${RED}WARNING${NC}] Given Input Is Not An Integer!"
+      echo ""
+    fi
+  done
+}
+
+###################################################
+# Name: web_page_setup                            #
+#                                                 #
+# Description: Runs web_page_chooser and modifies #
+# the location of html file in wifiphisher to user#
+# selected folder.                                #
+#                                                 #
+# Arguments: None                                 #
+#                                                 #
+# Returns: None                                   #
+###################################################
+function web_page_setup()
+{
+  # get the user's choice
+  web_page_chooser
+  readonly local user_choice=$?
+
+  # prompt user to make a backup file of wifiphisher.py file
+  read -p "$( echo -e "[${LIGHT_BLUE}INFO${NC}] Do You Want To Backup Your"\
+  "Current Settings? [Y/n]" )"
+  echo ""
+
+  if [[ "$REPLY" =~ ^[Yy]$ ]]; then
+    # make a backup of users current wifiphisher.py
+    echo -e "[${LIGHT_BLUE}INFO${NC}] Backing Up Your Current Wifiphisher.py File"
+    cp "${CURRENT_DIR}/wifiphisher.py" "${CURRENT_DIR}/wifiphisher.py.backup"
+    echo -e "[${LIGHT_BLUE}INFO${NC}] The wifiphisher.py.backup Has Been Created"
+    echo ""
+  fi
+
+  # the user selected files path
+  local new_path="web-pages/${FOLDERS[$user_choice]}"
+  echo -e "[${LIGHT_BLUE}INFO${NC}] Changing Files Now"
+  echo ""
+  # update the Wifiphisher.py file with the new index file
+  sed -i "28s%\(\"\).*\(\"\)%\1$new_path\2%" wifiphisher.py
```

Comment:
Another reason for this to be in Python is to replace these ugly hacks and directly assign the new path to the constant.